ledger/ledgercore/accountdata.go

```diff
@@ -134,6 +145,19 @@ func (u *AccountData) ClearOnlineState() {
 	u.VotingData = VotingData{}
 }
 
+// Suspend sets the status to Offline, but does _not_ clear voting keys, so
+// that a heartbeat can bring the account back Online
+func (u *AccountData) Suspend() {
+	u.Status = basics.Offline
+	// To regain eligibility, the account will have to `keyreg` with the extra fee.
+	u.IncentiveEligible = false
+}
+
+// Suspended returns true if the account is suspended (offline with keys)
+func (u *AccountData) Suspended() bool {
```

Comment:
Are no longer valid keys for offline accounts wiped out each round somewhere, or deleted incrementally regardless of online state?

Reply:
That is an excellent question, and I have no idea how deletion is managed.  I'll need to make sure nothing deletes them automatically when you go offline.  I suspect they are deleted only for being passed their round, because I know it is an expected case that you install part keys to a node _before_ it goes online, in anticipation of the `keyreg` being prepared elsewhere and sent.

Reply:
Found the deletion code elsewhere, its interesting.